Modules/_testbuffer.c

```diff
@@ -2823,63 +2823,86 @@ PyInit__testbuffer(void)
     PyObject *m;
 
     m = PyModule_Create(&_testbuffermodule);
-    if (m == NULL)
+    if (m == NULL) {
         return NULL;
+    }
 
     Py_SET_TYPE(&NDArray_Type, &PyType_Type);
     Py_INCREF(&NDArray_Type);
-    PyModule_AddObject(m, "ndarray", (PyObject *)&NDArray_Type);
+    if (PyModule_AddObject(m, "ndarray", (PyObject *)&NDArray_Type) < 0) {
+        Py_DECREF(&NDArray_Type);
+        return NULL;
```

Comment:
Note that it still leaks `m`.

It is more convenient to split the `PyInit_*` function on two parts: one creates the module object, calls the other function, and destroys the module object in case of error. The other function takes the module object as argument and returns -1 in case of error.